infra/modules/azure_event_hub/main.tf

```diff
@@ -0,0 +1,65 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.111.0"
+    }
+  }
+
+  required_version = "~> 1.7.5"
+}
+
+provider "azurerm" {
+  features {}
+
+  storage_use_azuread = true
+}
+
+module "naming_convention" {
+  source = "../azure_naming_convention"
+
+  environment = {
+    prefix          = var.environment.prefix
+    env_short       = var.environment.env_short
+    location        = var.environment.location
+    domain          = var.environment.domain
+    app_name        = var.environment.app_name
+    instance_number = var.environment.instance_number
+  }
+}
+
+#-----------------#
+# Azure Event HUB #
+#-----------------#
+
+module "event_hub" {
```

Comment:
I suggest to avoid any dependency to `terraform-azurerm-v3`

Reply:
Since we're not using anything from the original module, I agree in this specific case.